src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -297,6 +292,75 @@ def queue_privacy_request_to_propagate_consent(
     return privacy_request_results
 
 
+def _save_consent_preferences(
+    provided_identity: ProvidedIdentity,
+    db: Session,
+    consent_preference_data: List[ConsentSchema],
+) -> List[Consent]:
+    """Upsert Consent Preferences
+
+    For the old workflow: Updated existing Consent record with same data_use and provided identity. Otherwise, create a new one.
+    For the new workflow: Update an Existing Consent Record associated with the same Privacy Notice History record and provided identity.
+    Otherwise create a new one.
+
+    """
+    upserted_consent_preferences: List[Consent] = []
+
+    if any(pref.data_use for pref in consent_preference_data) and any(
+        pref.privacy_notice_id for pref in consent_preference_data
+    ):
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="Request has consent preferences saved for both data uses and privacy notices.  Migrate to using privacy notices.",
+        )
+
+    for preference in consent_preference_data:
+        current_preference: Optional[Consent] = None
+        privacy_notice_history: Optional[PrivacyNoticeHistory] = None
+
+        # Old workflow; slated to be deleted
+        if preference.data_use:
+            current_preference = Consent.filter(
+                db=db,
+                conditions=(Consent.provided_identity_id == provided_identity.id)
+                & (Consent.data_use == preference.data_use),
+            ).first()
+        # New workflow; preferences saved with respect to privacy notices
+        elif preference.privacy_notice_id and preference.privacy_notice_version:
```

Comment:
To me, it made sense for a user to save their preferences with respect to a privacy notice id and its current version, as the UI would showing notices from the privacy notices endpoints.  Then behind the scenes I link their Consent preference to the associated PrivacyNoticeHistory, and also put the contents of that PrivacyNoticeHistory on the privacy request.

It's a little bit of duplication here, but having the historical notice contents on the privacy request will help with reporting later, plus we don't have to do any extra querying when we go to propagate the preferences, we're using the notice saved on the privacy request.

(This is similar to what we do on main now, we copy the preferences onto the privacy request too w.r.t. data_use. )